pkg/purl/purl.go

```diff
@@ -198,13 +198,45 @@ func (p *PackageURL) BOMRef() string {
 	return purl.String()
 }
 
+// OverwritePkgIdentifiers overwrites package identifiers on the packages available
+// on the provided PackageInfo list adding missing OS metadata to existing PURLs
+// This is useful to overwrite identifiers for packages added by pkg (apk, rpm, etc.) analyzers
+func OverwritePkgIdentifiers(pkgInfos []ftypes.PackageInfo, os ftypes.OS) []ftypes.PackageInfo {
+	if os.Family == "" {
+		return pkgInfos
+	}
+
+	metadata := types.Metadata{
+		OS: &os,
+	}
+	for i, pkgInfo := range pkgInfos {
+		for j, pkg := range pkgInfo.Packages {
+			mewIdentifier := NewPackageIdentifier(os.Family, metadata, pkg)
+			pkgInfos[i].Packages[j].Identifier.PURL = mewIdentifier.PURL
+		}
+	}
+	return pkgInfos
+}
```

Comment:
I think we need to mode this function to new handler.
`purl` package generates purls, but it shouldn't overwrite fanal structure.